Runtime/CesiumCameraController.cs

```diff
@@ -0,0 +1,1145 @@
+        /// at the specified yaw and pitch.
+        /// </summary>
+        /// <remarks>
+        /// The characteristics of the flight can be configured with 
+        /// <see cref="CesiumCameraController.flyToAltitudeProfileCurve"/>,
+        /// <see cref="CesiumCameraController.flyToProgressCurve"/>, 
+        /// <see cref="CesiumCameraController.flyToMaximumAltitudeCurve"/>,
+        /// <see cref="CesiumCameraController.flyToDuration"/>, and
+        /// <see cref="CesiumCameraController.flyToGranularityDegrees"/>.
+        /// </remarks>
+        /// <param name="destination">The longitude (x), latitude (y), and height (z) of the destination.</param>
+        /// <param name="yawAtDestination">The yaw of the camera at the destination.</param>
+        /// <param name="pitchAtDestination">The pitch of the camera at the destination.</param>
+        /// <param name="canInterruptByMoving">Whether the camera flight can be interrupted with movement inputs.</param>
+        public void FlyToLocationLongitudeLatitudeHeight(
+            Vector3 destination,
+            float yawAtDestination,
+            float pitchAtDestination,
+            bool canInterruptByMoving)
+        {
+            double3 destinationCoordinates = new double3()
+            {
+                x = destination.x,
+                y = destination.y,
+                z = destination.z
+            };
+            double3 destinationECEF =
+                CesiumTransforms.LongitudeLatitudeHeightToEarthCenteredEarthFixed(
+                    destinationCoordinates);
+
+            this.FlyToLocationEarthCenteredEarthFixed(
+                destinationECEF,
+                yawAtDestination,
+                pitchAtDestination,
+                canInterruptByMoving);
+        }
+
+        #endregion
+
+        #region Dynamic clipping plane adjustment
+
+        private void UpdateClippingPlanes()
+        {
+            if (this._camera == null)
+            {
+                return;
+            }
+
+            // Raycast from the camera to the Earth's center and compute the distance.
+            float height = 0.0f;
+            if (!this.RaycastTowardsEarthCenter(out height))
+            {
+                return;
+            }
+
+            float nearClipPlane = this._initialNearClipPlane;
+            float farClipPlane = this._initialFarClipPlane;
+
+            if (height >= this._dynamicClippingPlanesMinHeight)
+            {
+                farClipPlane = height + (float)(2.0 * CesiumEllipsoid.GetMaximumRadius());
+                farClipPlane = Mathf.Min(farClipPlane, this._maximumFarClipPlane);
+
+                float farClipRatio = farClipPlane / this._maximumNearToFarRatio;
+
+                if (farClipRatio > nearClipPlane)
+                {
+                    nearClipPlane = Mathf.Min(farClipRatio, this._maximumNearClipPlane);
+                }
+                else
+                {
+                    nearClipPlane = 10.0f;
```

Comment:
Is this right? Why 10.0f rather than whatever the initial near plane was? If the initial near plane happened to be large, we could end up in this else block even though a near plane of 10.0f would cause Unity to cut off things in the distance because of its 100,000:1 far:near requirement.

Reply:
Ah thanks for the catch. This is leftover from when I was trying out different approaches / values to make it work; the code here got messy